include/vcpkg/vcpkgcmdarguments.h

```diff
@@ -118,6 +118,9 @@ namespace vcpkg
 
         static void append_common_options(HelpTableFormatter& target);
 
+        void set_manifest_overlays(std::vector<std::string>&, std::vector<std::string>&);
+        void merge_overlays();
```

Comment:
In my opinion, the overlay arrays should be merged at callsites using free functions and existing calls of `args.overlay_ports` and `args.overlay_triplets` should be replaced.

```
auto overlay_ports = merge_overlay_ports(args.overlay_ports, 
    args.env_overlay_ports, 
    paths.get_configuration().overlay_ports);
```

In the case of overlay ports it could be handled inside by changing it to `make_overlay_provider(VcpkgPaths&, const VcpkgCmdArguments&)` and reading the overlay arrays directly.

Reply:
I agree that the merging could be done in a different place but I worry that this approach would be more error prone, meaning that in the future every time someone uses args.overlay_ports they would have to remember to do a merge beforehand, otherwise you would lose the overlays from config and environment... Right now we don't have to worry about that because the code merges them automatically for us... what do you think?

Reply:
In this case, I would advise to make `VcpkgPaths` the container for overlay locations, an instance of `VcpkgPaths` is already available at every possible call-site and logically it makes more sense for overlay paths to be contained in the class with "paths" in its name. Also removes the need to remove `const` from `VcpkgCmdArguments&`.

Reply:
The problem is we want to avoid making VcpkgPaths an even bigger ball of mud than it already is. The name "paths" reflects that it really doesn't have a clear responsibility, and everywhere you need one the code becomes effectively untestable.

For example, the installed tree data structures used to live in there too but have since been factored out to be more testable.